### PR TITLE
Restructuring the test environment to run against PaaS and Verify

### PR DESCRIPTION
**Pull request by DavidJeche**
# Why I am making this change

This PR is about making sure that we are able to still perform tests for the different environments that we have. We need to make sure we test both verify and the PaaS modules, which take different arguments while running in completely two different networks.

We wanted to make sure that there are no clashes when running tests. We added a new environments file that we can use to make sure we when we run the tests in a way that is always unique. This also prevents clashes from happening when the running tests.

# What approach I took

We had to remove security group testing from the PaaS section as this is not currently supported. We split the test into two environments. When you run kitchen test there are two tests One is for verify and the other is for the PaaS. The tests have been split between the two space. They are unique per environment.

### CODE REVIEW

**Comment by kentsanggds** [comment]
Would be a good idea to include some instructions here as to how to run the tests, and it looks like the README will need updating to include this information as well.

**Comment by kentsanggds** [comment]
When running `kitchen converge` I get the following error - 

```
       Error: Error applying plan:
       
       2 error(s) occurred:
       
       * module.prometheus.aws_iam_policy.prometheus_instance_profile: 1 error(s) occurred:
       
       * aws_iam_policy.prometheus_instance_profile: Error creating IAM policy prometheus_instance_profile_prometheus-paas: EntityAlreadyExists: A policy called prometheus_instance_profile_prometheus-paas already exists. Duplicate names are not allowed.
       	status code: 409, request id: 8ee65811-bb23-11e8-9867-91687e38a6b5

       * module.prometheus.aws_iam_role.prometheus_role: 1 error(s) occurred:
       
       * aws_iam_role.prometheus_role: Error creating IAM Role prometheus_profile_prometheus-paas: EntityAlreadyExists: Role with name prometheus_profile_prometheus-paas already exists.
       	status code: 409, request id: 8d8baa0f-bb23-11e8-b310-eb54c2f666cb
```

**Comment by DavidJeche** [comment]
@kentsanggds Yep the readme does need some updating. The overall intention was to enable kitchen to run based on the variable passed into the environment. There is now an environment file that can be used to customize the test environment and values.

**Comment by DavidJeche** [comment]
> When running `kitchen converge` I get the following error -
> 
> ```
>        Error: Error applying plan:
>        
>        2 error(s) occurred:
>        
>        * module.prometheus.aws_iam_policy.prometheus_instance_profile: 1 error(s) occurred:
>        
>        * aws_iam_policy.prometheus_instance_profile: Error creating IAM policy prometheus_instance_profile_prometheus-paas: EntityAlreadyExists: A policy called prometheus_instance_profile_prometheus-paas already exists. Duplicate names are not allowed.
>        	status code: 409, request id: 8ee65811-bb23-11e8-9867-91687e38a6b5
> 
>        * module.prometheus.aws_iam_role.prometheus_role: 1 error(s) occurred:
>        
>        * aws_iam_role.prometheus_role: Error creating IAM Role prometheus_profile_prometheus-paas: EntityAlreadyExists: Role with name prometheus_profile_prometheus-paas already exists.
>        	status code: 409, request id: 8d8baa0f-bb23-11e8-b310-eb54c2f666cb
> ```

Ok, I can see that there is profile created on the 2018-09-18 10:14 UTC+0100. This appears to be today. I believe someone else may have tried to run it under the same name or it run but there were missing environment variables. 

I will update the readme inline to decribe the runtime init process. I believe the confusion here was caused by the two `kitchen list` options.

Please allow me some time to update the readme.

**Comment by kentsanggds** [comment]
I've noticed that `.../test/prometheus-paas/main.tf` and `.../test/prometheus-verify/main.tf` have very different structures. They should they be updated so that they are a bit more similar so that it will be easier to update the tests when things change in the main code.

File: terraform/modules/enclave/prometheus/test/integration/verify/inspec.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+name: Prometheus_Observe_GDS
+title: Test prometheus EC2 module
```
[the remark this answers is not in the document]
Comment (DavidJeche) [reply terraform/modules/enclave/prometheus/test/integration/verify/inspec.yml]: I have extended the titles however they provide not much benefit in the actual running tests. It helps with the clarity I suppose.

File: terraform/modules/enclave/prometheus/enviroment.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export TF_VAR_target_test_env="prometheus-paas"
```
[the remark this answers is not in the document]
Comment (DavidJeche) [reply terraform/modules/enclave/prometheus/enviroment.sh]: done

**Comment by philandstuff** [comment]
What's the status of this PR? The travis checks are failing - are you planning to fix that or do you want a review before then?

**Comment by kentsanggds** [comment]
I'm in the middle of fixing it and rebasing it on changes from master, almost there, it just takes a bit of time to run the tests.

**Comment by ejrowley** [comment]
I am getting the error `Error: Required variable not set: test_user` when I try and run kitchen test`

**Comment by kentsanggds** [comment]
Forgot to mention that you need to source `environment-test.sh`, it's documented in the enclave `README` to get the test running

**Pull request merged**